include/rmm/aligned.hpp

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2020-2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cassert>
+#include <cstddef>
+
+namespace rmm {
+
+/**
+ * @brief Default alignment used for host memory allocated by RMM.
+ *
+ */
+static constexpr std::size_t RMM_DEFAULT_HOST_ALIGNMENT{alignof(std::max_align_t)};
+
+/**
+ * @brief Default alignment used for CUDA memory allocation.
+ *
+ */
+static constexpr std::size_t CUDA_ALLOCATION_ALIGNMENT{256};
+
+/**
+ * @brief Returns whether or not `n` is a power of 2.
+ *
+ */
+constexpr bool is_pow2(std::size_t value) { return (0 == (value & (value - 1))); }
```

Comment:
Right. And align_up(x, 0) is not very useful. Note due to clang-tidy complaints this will be fixed with the less satisfying `return (value != 0U) && ((value & (value - 1)) == 0U);`

Note TIL in C++20 we can replace `is_pow2` with `std::has_single_bit`